File: Source/Core/Utility/NSManagedObjectContext+Utility.m
//
//  NSManagedObjectContext+Utility.m
//  Eunomia
//
//  Created by Ian Grossberg on 5/28/14.
//

#import "NSManagedObjectContext+Utility.h"

#import "NSLogWrapper.h"

@implementation NSManagedObjectContext (Eunomia_Utility)

- (BOOL)saveAndLogError:(NSString *)usageContext
{
    NSError *error;
    [self save:&error];
    if (error)
    {
        [NSLogWrapper error:@"%@: %@", usageContext, error];
    }
    
    return error != nil;
}

@end

File: Source/Core/UtilityExtensions/CoreData/NSManagedObjectContext+Utility.swift
//
//  NSManagedObjectContext+Utility.swift
//  Eunomia
//
//  Created by Ian Grossberg on 7/24/15.
//  Copyright (c) 2015 Adorkable. All rights reserved.
//

import Foundation

import CoreData

extension NSManagedObjectContext
{
    public func saveOrLogError(_ logContext : String) -> Bool
    {
        let result : NSError?
        do {
            
            try self.save()
            result = nil
        } catch let error as NSError {
            
            Log.error(message: "\(logContext): \(error)")
            result = error
        }
        
        return result == nil
    }
}
